src/core/agents.jl

```diff
@@ -414,67 +253,77 @@ function compute_base_fields(base_type_spec)
-end
+###########################################################################################
+# @multiagent
+###########################################################################################
 
 """
-    @dispatch f(args...)
+    @multiagent YourAgentType(AgentTypesToMerge) [<: OptionalSupertype]
 
-A macro to enable multiple-dispatch-like behavior for the function `f`,
-for various agent kinds generated via the [`@multiagent`](@ref) macro.
-For an illustration of its usage see the [Tutorial](@ref).
+Define multiple agent "subtypes", which are variants of a unique type `YourAgentType`. 
+This means that all "subtypes" are enclosed in the overarching type. Then, You cannot
+distinguish them on the basis of `typeof`, but need to use instead the `variantof`
+function. The `allvariants` function for a convenient way to obtain all variants types.
 
-If you are creating your own module/package that uses Agents.jl, and you 
-are using `@dispatch` inside it, then you need to put [`@finalize_dispatch`](@ref)`()`
-before the module end (but after all `@dispatch` calls).
-"""
-macro dispatch(f_def)
-    return esc(:(DynamicSumTypes.@pattern $f_def))
-end
-
-"""
-    @finalize_dispatch
-
-A macro to finalize the definitions of the methods generated by the 
-`@dispatch` macro when used in a module. It just needs to be used
-once at the end of the module if no `@dispatch` method is used inside
-of it. Otherwise use it also before the invocations of the methods.
+See the [Tutorial](@ref) or the [performance comparison versus `Union` types](@ref multiagent_vs_union)
+for why it is often better to use `@multiagent` than making multiple agent types. 
 
 ## Examples
-
-```julia
-module SomeModel
-
-@multiagent struct MultiAgent(NoSpaceAgent)
-    @subagent SubAgent1 end
-    @subagent SubAgent2 end
+Let's say you have this definition:
+```
+@agent struct Wolf
+    energy::Float64 = 0.5
+    ground_speed::Float64
+    const fur_color::Symbol
+end
+@agent struct Hawk{T}
+    energy::Float64 = 0.1
+    ground_speed::Float64
+    flight_speed::T
 end
 
-@dispatch MethodSub1(::SubAgent1) = 1
-@dispatch MethodSub1(::SubAgent2) = 1
+@multiagent Animal(Wolf, Hawk{Float64})
+```
+
+Then you can create `Wolf` and `Hawk` agents like so
+```
+hawk_1 = Animal'.Hawk(1, (1, 1), 1.0, 2.0, 3)
+hawk_2 = Animal'.Hawk(; id = 2, pos = (1, 2), ground_speed = 2.3, flight_speed = 2)
+wolf_1 = Animal'.Wolf(3, (2, 2), 2.0, 3.0, :black)
+wolf_2 = Animal'.Wolf(; id = 4, pos = (2, 1), ground_speed = 2.0, fur_color = :white)
+```
 
-@finalize_dispatch()
+The way to retrieve the variant of the agent is through the function `variantof` e.g.
+```
+variantof(hawk_1) # Hawk
+variantof(wolf_2) # Wolf
+```
 
-end
+You can also access the enclosed variant instance with the variant `function`
+```
+variant(hawk_1) # Hawk(1, (1, 1), 1.0, 2.0, 3.0)
+variant(wolf_1) # Wolf(3, (2, 2), 2.0, 3.0, :black)
 ```
+
+See the [rabbit_fox_hawk](@ref) example to see how to use this macro in a model.
```

Comment:
The hyperlink here is broken.

Reply:
I'm not sure how to fix it quickly because I'm a bit rusty on Documenter things, could you fix it if you know how? I addressed the rest of the review

Reply:
I can't checkout the branch now but the fix is easy: first, change the title of the example to:
```
# [# 3D Mixed-Agent Ecosystem with Pathfinding](@id rabbit_fox_hawk)
```
in the example `.jl` file. Then refer to it as `[rabbit fox hawk example](@ref rabbit_fox_hawk)`.